clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/config/ProviderHelpers.java

```diff
@@ -209,14 +211,32 @@ public static BuildResult buildAwsProviderAgents(
         }
       }
     }
+    if (reservationReportPool.isPresent() && awsProvider.getAgentScheduler() == null) {
```

Comment:
What makes awsProvider.getAgentScheduler() not null ... I'm seeing the reservation report caching agent not running but haven't had time to dig yet.

Reply:
I wanted to leave as much code as it was written originally but I now realize checking for null is not a good idea here due to timing. I will do an explicit check for the presence of `ReservationReportCachingAgent` once, since it needs to run across all accounts in repository.

Reply:
Fixed this issue and `InstanceTerminationLifeCycleWorker` issue. Please verify and thank you for your help.